classes/Exception/ThankYouOClass.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Claromentis\ThankYou\Exception;
+
+class ThankYouOClass extends ThankYouException
```

Comment:
`UnsupportedOwnerClassException` might be clearer

Reply:
I decided to instead create two Exceptions (`OwnerClassNameException` and `UnsupportedThankYouOwnerClass`), as the original exception was used to define a couple of different problems.

Fixed in a982e677dc7c426c64e8f27dd4e43fdae27d5890

Worth a review as I've deviated from your suggestion.

Reply:
I imagine the latter lacks an `Exception` suffix because it's already a long class name, but I'd put it in there anyway for consistency. Otherwise seems fine.

Reply:
Also, the main reason I suggested `UnsupportedOwnerClassException` instead of `UnsupportedThankYouOwnerClassException`, other than for brevity, is because the namespace already contains this information:

```
Claromentis\ThankYou\Exception\UnsupportedOwnerClassException
```

We don't do this for suffixes (`Controller`, `Exception`) because these distinguish the base class name by type. The namespace can be used to distinguish what module it's for (and there's no other exception with this name anyway).

Reply:
I've renamed the Exception to `UnsupportedOwnerClassException` as requested.

The implementation of the `OwnerClassNameException` allows the Factory to do the substitution of the Owner Class Name in the method `GetOwnerClassName`. I think this is much cleaner than having the underlying Utility method make such a decision, an indeed has the same intended affect. Happy to discuss further if needs be :)

Reply:
Why does that need to be an exception, then? You could just return an empty string and use the localised message if an empty string is returned.

Reply:
Like so:

```php
/**
 * @param int $owner_class_id
 * @return string
 */
private function GetOwnerClassName(int $owner_class_id): string
{
	return $this->utility->GetOwnerClassName($owner_class_id) ?: ($this->lmsg)('thankyou.owner_class.not_found');
}
```

Reply:
Also, it looks like there's an unused `GetOwnerClassNames` method on the API that goes straight to the utility. That could be removed.